gymnasium/envs/mujoco/mujoco_rendering.py

```diff
@@ -134,14 +139,48 @@ def close(self):
 class OffScreenViewer(BaseRender):
     """Offscreen rendering class with opengl context."""
 
-    def __init__(self, model: "mujoco.MjMujoco", data: "mujoco.MjData"):
-        width = model.vis.global_.offwidth
-        height = model.vis.global_.offheight
+    def __init__(
+        self,
+        model: "mujoco.MjMujoco",
+        data: "mujoco.MjData",
+        width: Optional[int] = None,
+        height: Optional[int] = None,
+        max_geom: int = 1000,
+    ):
+        buffer_width = model.vis.global_.offwidth
+        buffer_height = model.vis.global_.offheight
+
+        width = width or buffer_width
+        height = height or buffer_height
+
+        # check if the framebuffer is large enough to handle the requested image dimensions.
+        # same check as in `mujoco.Renderer` class
+        if width > buffer_width:
```

Comment:
Remove the buffer checks as they are not applicable to all possible MuJoCo environments

Reply:
fixed in d30cd50